Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/SettingsResetFragment.kt

```diff
@@ -52,46 +50,22 @@ class SettingsResetFragment : Fragment(R.layout.fragment_settings_reset) {
         binding.settingsResetContainer.sendAccessibilityEvent(AccessibilityEvent.TYPE_ANNOUNCEMENT)
     }
 
-    private fun deleteAllAppContent() {
-        lifecycleScope.launch {
-            try {
-                val isTracingEnabled = InternalExposureNotificationClient.asyncIsEnabled()
-                // only stop tracing if it is currently enabled
-                if (isTracingEnabled) {
-                    InternalExposureNotificationClient.asyncStop()
-                    BackgroundWorkScheduler.stopWorkScheduler()
-                }
-            } catch (apiException: ApiException) {
-                apiException.report(
-                    ExceptionCategory.EXPOSURENOTIFICATION, TAG, null
-                )
-            }
-            withContext(Dispatchers.IO) {
-                deleteLocalAppContent()
-            }
-            navigateToOnboarding()
-        }
-    }
-
     private fun navigateToOnboarding() {
         OnboardingActivity.start(requireContext())
         activity?.finish()
     }
 
-    private fun deleteLocalAppContent() {
-        DataRetentionHelper.clearAllLocalData(requireContext())
-    }
-
     private fun confirmReset() {
         val resetDialog = DialogHelper.DialogInstance(
             requireActivity(),
             R.string.settings_reset_dialog_headline,
             R.string.settings_reset_dialog_body,
             R.string.settings_reset_dialog_button_confirm,
             R.string.settings_reset_dialog_button_cancel,
-            true,
-            {
-                deleteAllAppContent()
+            cancelable = true,
```

Comment:
Launching something and deciding in which scope or on what thread is the viewmodels job, not the fragments.

Reply:
Ok. I get what you mean.  I was doing this because deleteAllAppContent() was a suspend function but now I changed it so I can call it normally from the fragment.